### PR TITLE
[Impl](Nereids) add nereids gc cost time and be fold const cost time to profile

### PR DESCRIPTION
**Pull request by LiBinfeng-01**
add Nereids  GarbageCollect  Time and Nereids  BeFoldConst  Time to profile

### CODE REVIEW

**Comment by LiBinfeng-01** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/NereidsPlanner.java [hunk 1]
```diff
@@ -536,6 +543,15 @@ private PhysicalPlan chooseBestPlan(Group rootGroup, PhysicalProperties physical
...
     }
 
+    private long getGarbageCollectionTime() {
+        List<GarbageCollectorMXBean> gcMxBeans = ManagementFactory.getGarbageCollectorMXBeans();
+        long initialGCTime = 0;
+        for (GarbageCollectorMXBean gcBean : gcMxBeans) {
+            initialGCTime += gcBean.getCollectionTime();
+        }
+        return initialGCTime;
+    }
```
Comment (924060929) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/NereidsPlanner.java:553]: How long does this function usually take?
If > 10ms, we should only compute this time when SessionVariable.enable_profile == true

Comment (LiBinfeng-01) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/NereidsPlanner.java:553]: it should be test on complex condition, it takes 0ms in local test

**Comment by py023** [comment]
run p0

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run p0

**Review by 924060929 — DISMISSED** [review]

**Comment by LiBinfeng-01** [comment]
run buildall

**Review by 924060929 — APPROVED** [review]

**Comment by LiBinfeng-01** [comment]
run p0

**Comment by LiBinfeng-01** [comment]
run p0

**Comment by LiBinfeng-01** [comment]
run p0

**Comment by englefly** [comment]
if this query is running during a gc, the whole gc time is counted, right?

**Review by englefly — APPROVED** [review]

**Review by hello-stephen — APPROVED** [review]

**Pull request merged**